rust/src/treefile.rs

```diff
@@ -2260,3 +2271,45 @@ pub(crate) fn treefile_new_client(filename: &str, basearch: &str) -> CxxResult<B
     r.error_if_base()?;
     Ok(r)
 }
+
+fn iter_etc_treefiles() -> Result<impl Iterator<Item = Result<PathBuf>>> {
+    Ok(read_dir(CLIENT_TREEFILES_DIR)?.filter_map(|res| match res {
+        Ok(e) => {
+            let path = e.path();
+            if let Some(ext) = path.extension() {
+                if ext == "yaml" {
+                    return Some(anyhow::Result::Ok(path));
+                }
+            }
+            None
+        }
+        Err(err) => Some(Err(anyhow::Error::msg(err))),
+    }))
+}
+
+/// Create a new client treefile using the treefile dropins in /etc/rpm-ostree/origin.d/.
+pub(crate) fn treefile_new_client_from_etc(basearch: &str) -> CxxResult<Box<Treefile>> {
+    let basearch = opt_string(basearch);
+    let mut cfg = TreeComposeConfig::default();
+    let mut tfs = iter_etc_treefiles()?.collect::<Result<Vec<PathBuf>>>()?;
+    tfs.sort(); // sort because order matters; later treefiles override earlier ones
```

Comment:
Oh right, that's a trap.  Hmm.  Also an argument for having the API for loading config files return a `Vec<>` directly.  But that can be a followup.

Or...I dunno actually, the more I think about this the more I think we shouldn't allow overrides at all?  Allowing it just seems like it's most likely to be a footgun for users.

At least this isn't a problem yet right because we're only supporting packages which are always unionioned?